dojo/tools/checkov/parser.py

```diff
@@ -15,32 +15,41 @@ def get_description_for_scan_types(self, scan_type):
         return "Import JSON reports of Infrastructure as Code vulnerabilities."
 
     def get_findings(self, json_output, test):
+        findings = list()
+        if json_output:
+            report = self.parse_json(json_output)
+            for tree in report:
+                check_type = tree['check_type']
+                findings += self.get_items(tree, test, check_type)
 
-        if json_output is None:
-            return list()
-
-        tree = self.parse_json(json_output)
-
-        check_type = ''
-        if 'check_type' in tree:
-            check_type = tree['check_type']
-
-        if tree:
-            return [data for data in self.get_items(tree, test, check_type)]
-        else:
-            return list()
+        return findings
 
     def parse_json(self, json_output):
+        """Parse JSON report.
+        Checkov may return only one `check_type` (where the report is just a JSON)
+        or more (where the report is an array of JSONs).
+        To address all scenarios we force this method to return a list of JSON objects.
+
+        :param json_output: JSON report
+        :type json_output: file
+        :return: JSON array of objects
+        :rtype: list
+        """
         try:
             data = json_output.read()
             try:
-                tree = json.loads(str(data, 'utf-8'))
+                report = json.loads(str(data, 'utf-8'))
             except:
-                tree = json.loads(data)
+                report = json.loads(data)
         except:
             raise Exception("Invalid format")
 
-        return tree
+        if type(report) is not list:
+            tmp = list()
+            tmp.append(report)
+            report = tmp
```

Comment:
I'm not super happy about this part but couldn't find a pythonic alternative. I mean, it works properly, but it looks ugly idk. Any suggestions?

Reply:
You could rename the variables in this method. If you change `report` to something like `deserialized` and `tmp` to `nodes` it would look much friendlier.

Reply:
```Python
if type(report) is not list:
    return [report]
```

Reply:
Awesome, thank you guys ❤. Take a look at https://github.com/DefectDojo/django-DefectDojo/pull/4705/commits/0e67482830ec7acc8248e0044605ed7b973532e1, the code should look better now 😊